maven-core/src/main/java/org/apache/maven/internal/aether/MavenChainedWorkspaceReader.java

```diff
@@ -60,6 +62,31 @@ public Model findModel(Artifact artifact) {
         return null;
     }
 
+    @Override
+    public Stream<Artifact> listArtifacts() {
+        return Arrays.stream(readers)
+                .flatMap(r -> {
+                    // TODO once resolver 2.x is there this part can be simplified
+                    if (r instanceof MavenWorkspaceReader) {
+                        return ((MavenWorkspaceReader) r).listArtifacts();
+                    }
+                    return Stream.empty();
```

Comment:
personally I prefer if-else, reading code similar to this requires more attention IMHO and is easy to slip on it

Reply:
maybe use `filter(it -> instanceof MavenWorkspaceReader).map(MavenWorkspaceReader.class::cast)`, will avoid the debate of `if/else` (readability on the left vs symmetry - no good choice there)